judge/views/submission.py

```diff
@@ -531,6 +531,21 @@ def get(self, request, *args, **kwargs):
         return super(ForceContestMixin, self).get(request, *args, **kwargs)
 
 
+class UserAllContestSubmissions(ForceContestMixin, AllUserSubmissions):
+    def get_title(self):
+        return "%s's submissions in %s" % (self.username, self.contest.name)
+
+    def access_check(self, request):
```

Comment:
I think we should only open up this view for contest organizers, since otherwise this bypasses the checks that were put in place to stop users from seeing contest problem submissions outside of the contest.

Reply:
This uses the same access checks as `UserContestSubmission`

Reply:
`UserContestSubmissions` has `UserProblemSubmissions` in its inheritance tree, so it also checks for problem accessibility (making sure the user can access the problem before being able to view the related contest submissions). On the other hand, there is only `ForceContestMixin` and `AllUserSubmissions`, which means the only contest-specific is checking that the contest is running from `ForceContestMixin`.

Reply:
Hm so in this case, access should be granted if:
1. You can edit the contest in some form
2. Scoreboard is publicly accessible
3. It's your participation
?

Reply:
Currently, you can only access in-contest submissions (for a certain problem) if:
 1. You can edit the contest.
 2. You can access the problem _outside_ of the contest.
 3. You are in the contest.

I'm not sure whether this is the correct behaviour or not.